src/CommandRunner.cpp

```diff
@@ -0,0 +1,114 @@
+#include "CommandRunner.hpp"
+#include <memory>
+#include <stdio.h>
+#include <fstream>
+#include <sstream>
+#include <iostream>
+
+#include <stout/try.hpp>
+
+#define TEMP_FILE_TEMPLATE "/tmp/criteo-mesos-XXXXXX"
+
+namespace criteo {
+namespace mesos {
+namespace CommandRunner {
+
+/*
+ * Represent a temporary file that can be either written or read from.
+ */
+class TemporaryFile
+{
+public:
+  TemporaryFile() {
+    char filepath[] = TEMP_FILE_TEMPLATE;
+    if(mkstemp(filepath) == -1)
+      throw std::runtime_error("Unable to create temporary file");
```

Comment:
Well, in the logs you will have the filename of the source code anyway but I will still augment the log line because it might not be easy to know if this file is in Mesos or something else.